doc/ir-semantics.text

```diff
@@ -120,9 +120,9 @@ Update (partial) function, returns an H:
     where v := eval P H E e
   osr-env P H E M (mut x)       = (x ↦ ⊥)
   osr-env P H E M (mut x = &y)  = (x ↦ ref a)
-    where a = E(y)
-  osr-env P H E M (mut x = e)   = (x ↦ ref e)
-    where (a ↦ _) = M(x)
+    where E ∋ (y ↦ ref a)
+  osr-env P H E M (mut x = e)   = (x ↦ ref a)
+    where M ∋ (x ↦ (a ↦ _))
```

Comment:
I think this line should evaluate `e` first and put the result in the memory at `a`.